### ISSUE TITLE
Session Start scan guard false-alarms "SCAN COMMAND BROKEN" the moment its output is filtered — the `parsed` counter rides the stream it guards

### ISSUE THREAD

**Issue by dispather** [rebelytics/one-skill-to-rule-them-all#100]
## What happened

Running the Session Start Protocol step 2 scan snippet, I piped the loop's output to see only the fields step 2 actually needs:

```bash
for f in "$d"/*.md; do ... printf '%s\n---\n' "$hdr"; done | grep -E '^(id|title|status|skill):'
```

The pipe put the whole loop into a subshell, so `parsed` incremented only in the child. The parent read `parsed=0`, and the guard printed `SCAN COMMAND BROKEN — 17 files present, 0 headers parsed` — directly *below* 17 correctly-printed headers.

This is not probabilistic: anyone who filters the output hits it 100% of the time. And the skill treats that message as a stop signal ("a structural probe that comes back empty ... is a stop signal, not a create"), so taken literally, a healthy session goes hunting for a migration that never happened. Worse, the incentive to filter grows with the log: once the log holds hundreds of entries, dumping every frontmatter block to stdout is exactly what a user will want to trim — so the guard lies most often in the very regime it was built for.

(Related but distinct from #87: that report is about the guard's *exit status* on the success path; this one is about the guard's *verdict* being corrupted by any downstream pipe/filter on the loop output.)

## Why the existing text failed to prevent it

The snippet's guard is correct in isolation; nothing warns that the counter shares a compound command with the stream being emitted, so any downstream transformation (pipe/redirect/filter) silently changes the verdict. The failure direction in shell is always "0", which is the guard's *strongest* alarm.

## Proposed fix (either suffices)

1. Compute the integrity count where a pipe cannot strip it — e.g. a separate command (`parsed=$(grep -clE '^---' "$d"/*.md | wc -l)` style), or write the count to a temp file inside the loop and read it back after.
2. Better: stop dumping full frontmatter to stdout at all — emit only the fields step 2 needs. Then there is no reason to filter, and the rule needs no remembering.

Minimal textual mitigation if the snippet stays as-is: one line beside it — "keep the header dump and the counter out of one compound command; filtering the output traps the counter in a subshell and zeroes it."

General principle we derived from this (feel free to reuse): **a guard's state must not travel through the stream it guards** — otherwise the verdict becomes a function of the transport, not of the target.

Observed live on Linux (tool shell zsh, snippet run under bash semantics as documented) with a 17-entry log, 2026-08-30; reproduced trivially by appending any pipe to the loop.

**Comment by rebelytics** [comment]
Fixed in `main` and shipped in v3.1.0 today — the parsed count is derived by its own command, so piping the loop can't move it into a subshell; verified with your exact `done | grep` shape. Thanks for the report.
